RELEASE/scripts/autoscend/quests/level_any.ash

```diff
@@ -697,3 +697,120 @@ boolean LX_meatMaid()
+			return "auto";
+	}
+}
+
+boolean LX_setWorkshed(){
+	string desiredShed = LX_getDesiredWorkshed();
+	string existingShed = get_workshed();
+	string workshedChanged = get_property("_workshedItemUsed");
+
+	if (workshedChanged == "true") return false; //Don't even try if the workshed has already been changed once
+
+	//Check to make sure we can use the workshed item and that it isn't already in the campground. If already in campground, return false also
+	//These first 2 ifs are only used if something other than auto is specified
+	if (auto_is_valid(to_item(desiredShed)) && (contains_text(existingShed, desiredShed) == false) && (item_amount(to_item(desiredShed)) > 0))
+	{
+		use(1, to_item(desiredShed));
+		return true;
+	}
+	if (contains_text(existingShed, desiredShed))
+	{
+		return false;
+	}
+	//Auto workshed changing
+	if(contains_text(desiredShed,"auto"))
+	{
+		//Check if there is enough lumber for orcs. The existing shed check is because we only want to go into this
+		//if statement once to use the best available workshed
+		if((fastenerCount() < 30 && lumberCount() < 30) && (existingShed == "none"))
+		{
+			if ((auto_is_valid($item[model train set])) && (item_amount($item[model train set]) > 0))
+			{
+				use(1, $item[model train set]);
+				return true;
+			}
+			if ((auto_is_valid($item[Asdon Martin keyfob])) && (item_amount($item[Asdon Martin keyfob]) > 0))
+			{
+				use(1, $item[Asdon Martin keyfob]);
+				return true;
+			}
+			if ((auto_is_valid($item[cold medicine cabinet])) && (item_amount($item[cold medicine cabinet]) > 0))
+			{
+				use(1, $item[cold medicine cabinet]);
+				return true;
+			}
+			if ((auto_is_valid($item[little geneticist dna-splicing lab])) && (item_amount($item[little geneticist dna-splicing lab]) > 0))
+			{
+				use(1, $item[little geneticist dna-splicing lab]);
+				return true;
+			}
+			if ((auto_is_valid($item[portable mayo clinic])) && (item_amount($item[portable mayo clinic]) > 0))
+			{
+				use(1, $item[portable mayo clinic]);
+				return true;
+			}
+			return false;
+		}
+		if((fastenerCount() >= 30 && lumberCount() >= 30)) //once we have enough fasteners
+		{
+			if ((auto_is_valid($item[Asdon Martin keyfob])) && (item_amount($item[Asdon Martin keyfob]) > 0))
+			{
+				use(1, $item[Asdon Martin keyfob]);
+				return true;
+			}
+			if ((auto_is_valid($item[cold medicine cabinet])) && (item_amount($item[cold medicine cabinet]) > 0))
+			{
+					use(1, $item[cold medicine cabinet]);
+					return true;
+			}
+			if ((auto_is_valid($item[little geneticist dna-splicing lab])) && (item_amount($item[little geneticist dna-splicing lab]) > 0))
+			{
+				use(1, $item[little geneticist dna-splicing lab]);
+				return true;
+			}
+			if ((auto_is_valid($item[portable mayo clinic])) && (item_amount($item[portable mayo clinic]) > 0))
+			{
+				use(1, $item[portable mayo clinic]);
+				return true;
+			}
+			return false; //return false if no other workshed is available

```

Comment:
Nice to have - add debug log statement for not being able to find a workshed to switch to
`auto_log_debug("Unable to find workshed to install besides train");`
I'm not really sold on this wording. Feel free to adjust it!